Bug/73: Posting problems (DEMO)

jluukka opened the pull request:
- Allow only numeric characters, decimals and minus sign in numeric fields of posting grid
- Added onfocus handlers to select all text in an input field only when navigated to using the tabulator key

ilkkalehtinen commented on src/client/Components/GridColumns.js:
```diff
@@ -19,6 +19,16 @@ export const renderEditable = (cellInfo, data, updateData) => {
+      onKeyPress={e => {
+        // If numeric field, allow only number characters, decimal points and minus sign
+        let c = e.char || String.fromCharCode(e.charCode);
+        if (numeric && c.match(/[^0-9.,-]/gmi) !== null) {
```
The decimal separator varies according to locale. I18n service should be used to find out the proper separators. 

There might be a need for validation in onBlur as well e.g. highlighting cell with incorrect content.

jluukka commented:
- Added localized decimal characters to input validation on keypress
- Added validation to blur event make sure the new value is a proper numeric value, localization taken into account with the decimal character
